controllers/case-controller.js

```diff
@@ -0,0 +1,47 @@
+const createError = require('http-errors');
+
+const CaseController = Case => {
+  const createCase = async body => {
+    const caseRecord = {
+      info: body.info,
+      helpline: body.helpline,
+      status: body.status || 'open',
+    };
+
+    const createdCase = await Case.create(caseRecord);
+    return createdCase;
+  };
+
+  const getCase = async id => {
+    const caseFromDB = await Case.findByPk(id);
+
+    if (!caseFromDB) {
+      const errorMessage = `Case with id ${id} not found`;
+      throw createError(404, errorMessage);
+    }
+
+    return caseFromDB;
+  };
+
+  const listCases = async query => {
+    const queryObject = {
+      order: [['createdAt', 'DESC']],
+      where: {
+        helpline: query.helpline || '',
+      },
+    };
+
+    const cases = await Case.findAll(queryObject);
+    return cases;
+  };
+
+  const updateCase = async (id, body) => {
+    const caseFromDB = await getCase(id);
+    const updatedCase = await caseFromDB.update(body);
```

Comment:
Does this take the exact body from the request and place it in?

Specific example, if I have an existing object like:
```
{
  a: "foo",
  b: "bar"
}
```
and then I get a request:
```
{
  b: 'baz'
}
```
then is my resulting object:
```
{
  a: 'foo',
  b: 'baz'
}
```
or
```
{
  b: 'baz'
}
```

Reply:
It will only update the provided properties, and not replace the whole body. In your example, the resulting object will be: 
`{
  a: 'foo',
  b: 'baz'
}`